monai/application/datasets.py

```diff
@@ -120,3 +120,114 @@ def _generate_data_list(self, dataset_dir):
+        num_workers: the number of worker threads to use.
+            If 0 a single thread will be used. Default is 0.
+
+    Raises:
+        ValueError: root_dir must be a directory.
+        ValueError: unsupported task.
+        RuntimeError: can not find dataset directory, please use download=True to download it.
+
+    """
+
+    resource = {
+        "Task01_BrainTumour": "https://drive.google.com/uc?id=1A2IU8Sgea1h3fYLpYtFb2v7NYdMjvEhU",
+        "Task02_Heart": "https://drive.google.com/uc?id=1wEB2I6S6tQBVEPxir8cA5kFB8gTQadYY",
+        "Task03_Liver": "https://drive.google.com/uc?id=1jyVGUGyxKBXV6_9ivuZapQS8eUJXCIpu",
+        "Task04_Hippocampus": "https://drive.google.com/uc?id=1RzPB1_bqzQhlWvU-YGvZzhx2omcDh38C",
+        "Task05_Prostate": "https://drive.google.com/uc?id=1Ff7c21UksxyT4JfETjaarmuKEjdqe1-a",
+        "Task06_Lung": "https://drive.google.com/uc?id=1I1LR7XjyEZ-VBQ-Xruh31V7xExMjlVvi",
+        "Task07_Pancreas": "https://drive.google.com/uc?id=1YZQFSonulXuagMIfbJkZeTFJ6qEUuUxL",
+        "Task08_HepaticVessel": "https://drive.google.com/uc?id=1qVrpV7vmhIsUxFiH189LmAn0ALbAPrgS",
+        "Task09_Spleen": "https://drive.google.com/uc?id=1jzeNU1EKnK81PyTsrx0ujfNl-t0Jo8uE",
+        "Task10_colon": "https://drive.google.com/uc?id=1m7tMpE9qEcQGQjL_BdMD-Mvgmc44hG1Y",
+    }
+    md5 = {
+        "Task01_BrainTumour": "240a19d752f0d9e9101544901065d872",
+        "Task02_Heart": "06ee59366e1e5124267b774dbd654057",
+        "Task03_Liver": "a90ec6c4aa7f6a3d087205e23d4e6397",
+        "Task04_Hippocampus": "9d24dba78a72977dbd1d2e110310f31b",
+        "Task05_Prostate": "35138f08b1efaef89d7424d2bcc928db",
+        "Task06_Lung": "8afd997733c7fc0432f71255ba4e52dc",
+        "Task07_Pancreas": "4f7080cfca169fa8066d17ce6eb061e4",
+        "Task08_HepaticVessel": "641d79e80ec66453921d997fbf12a29c",
+        "Task09_Spleen": "410d4a301da4e5b2f6f86ec3ddba524e",
+        "Task10_colon": "bad7a188931dc2f6acf72b08eb6202d0",
+    }
+
+    def __init__(
+        self,
+        root_dir: str,
+        task: str,
+        section: str,
+        transform: Callable[..., Any],
+        download: bool = False,
+        seed: int = 0,
+        val_frac: float = 0.2,
+        cache_num: int = sys.maxsize,
+        cache_rate: float = 1.0,
+        num_workers: int = 0,
+    ):
+        if not os.path.isdir(root_dir):
+            raise ValueError("root_dir must be a directory.")
+        self.section = section
+        self.val_frac = val_frac
+        self.set_random_state(seed=seed)
+        if task not in self.resource:
+            raise ValueError(f"unsupported task: {task}.")
+        dataset_dir = os.path.join(root_dir, task)
+        tarfile_name = f"{dataset_dir}.tar"
+        if download:
+            download_and_extract(self.resource[task], tarfile_name, root_dir, self.md5[task])
+
+        if not os.path.exists(dataset_dir):
+            raise RuntimeError("can not find dataset directory, please use download=True to download it.")
```

Comment:
need more info for this error: `f"cannot find the directory {dataset_dir}"`....

Reply:
updated both MedNISTDataset and DecathlonDataset in the latest commit.
Thanks.